libs/s25main/PickedMovableObject.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "PickedMovableObject.h"
+#include "desktops/dskGameInterface.h"
+#include "drivers/VideoDriverWrapper.h"
+#include "world/GameWorldBase.h"
+#include "world/GameWorldView.h"
+#include "nodeObjs/noMovable.h"
+#include <cmath>
+
+namespace {
+constexpr unsigned PickRadius = 2;
+constexpr unsigned TrackRadius = 5;
+constexpr unsigned long ExpireIn = 5000;
+constexpr auto ExpireNever = static_cast<unsigned long>(-1);
```

Comment:
Maybe move this to the class and use it as the initializer? Avoids the comment for `expiration_`

Reply:
Yeah. Was thinking that too, but was reluctant to have constants in two places.

Reply:
Everything is better than magic numbers ;-)

Best code I've seen (by a big company):

```
def foo():
   """Create 2MB temporary file"""
  tempfile.make(FILE_SIZE_1MB)

FILE_SIZE_1MB = 1024 * 1024 * 8
```

TLDR: Documentation/comment is always out of date, so `/* = ExpireNever */` style is a clear indication to move that constant higher up.

Reply:
lol